src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java

```diff
@@ -50,43 +50,61 @@ public NotificationOperations(DatabaseManager manager, String username, String p
 
     /**
      * returns all the stored notifications
+     *
      * @return a list of notifications
      */
-    public List<NotificationRecord> getAllNotifications() {
+    public List<Notification> getAllNotifications() {
         return create.selectFrom(Tables.NOTIFICATION)
-                .fetch();
+                .fetch()
+                .map(this::fetchAndAddEmailsToNotification);
+    }
+
+
+    /**
+     * Returns a single notification.
+     *
+     * @param id the ID of the notification
+     * @return the notification of empty if not found
+     */
+    public Optional<Notification> getNotification(int id) {
+        return create.fetchOptional(NOTIFICATION, NOTIFICATION.ID_NOTIFICATION.eq(id))
+                .map(this::fetchAndAddEmailsToNotification);
     }
 
     /**
      * Returns a range of notifications starting from {@code cursor}
      *
      * @param cursor pagination cursor
-     * @param next {@code true} for next, {@code false} for previous
-     * @param limit the umber of records
+     * @param next   {@code true} for next, {@code false} for previous
+     * @param limit  the umber of records
      * @return a list of notifications
      */
     public Range<Notification, Integer> getNotificationsFrom(int cursor, boolean next, int limit) {
         return getNextRange(create.selectFrom(NOTIFICATION), NOTIFICATION.ID_NOTIFICATION, NOTIFICATION, cursor, next, limit)
-                .map(NotificationTransformer::toProto);
+                .map(this::fetchAndAddEmailsToNotification);
     }
 
     /**
-     * inserts a notification into the database.
-     * @param record the record to insert
-     * @return the resulting record (the primary key is guaranteed to be set)
+     * Fetches all NotificationReceiverEmailRecords for the given NotificationRecord and
+     * adds the emails to the protobuf.
+     *
+     * @param notificationRecord the NotificationRecord to add the emails to
+     * @return a Notification
      */
-    public NotificationRecord insertNotification(NotificationRecord record) {
-        return create.insertInto(Tables.NOTIFICATION)
-                .set(record)
-                .returning()
-                .fetchOne();
+    private Notification fetchAndAddEmailsToNotification(NotificationRecord notificationRecord) {
```

Comment:
i mean thats an optimisation, but there is also a mapList operation defined in getNextRange, so you could alter the signature to `fetchAndAddEmailsToNotifications(List<NotificationRecord> notificationRecords)` and the implementation to `create.....where(notification.id.in(notificationRecords)`.map(NotificationTransformer::toProto)`